ceph_cfg/mdl_updater.py

```diff
@@ -227,31 +223,77 @@ def partitions_all_refresh(self):
-            part_type = None
-            for line in output["stdout"].split('\n'):
-                split_line = line.split(':')
-                if split_line[0] != 'Partition Table':
+    def partitions_all_refresh_parted(self):
+        '''
+        List all partition details using parted
+        '''
+        arguments = [
+            util_which.which_parted.path,
+            '-s',
+            '-m',
+            '-l',
+            'print'
+            ]
+        output = utils.execute_local_command(arguments)
+        if output["retcode"] != 0:
+            raise Error("Failed executing '%s' Error rc=%s, stdout=%s stderr=%s" % (
+                " ".join(arguments),
+                output["retcode"],
+                output["stdout"],
+                output["stderr"]
+                ))
+        lines = output["stdout"].split('\n')
+        lines_num = len(lines)
+        if lines_num == 0:
+            return
+        chunk_lines = []
+        chunk_end = int(lines_num - 1)
+        for line_num in list(reversed(range(lines_num))):
+            if lines[line_num] == 'BYT;':
+                chunk_lines.append((int(line_num), chunk_end))
+                chunk_end = int(line_num)
+        parted_dict = {}
+        for chunk_start,chunk_end  in chunk_lines:
+            chunk_list = lines[chunk_start:chunk_end]
+            disk_line_split = chunk_list[1].split(':')
+            parted_dict_disk = {
+                'disk' : disk_line_split[0],
+                'size' : disk_line_split[1],
+                'driver' : disk_line_split[2],
+                'sector_size_logical' : disk_line_split[3],
+                'sector_size_physical' : disk_line_split[4],
+                'table' : disk_line_split[5],
+                'vendor' : disk_line_split[6],
+                'partition' : {}
+                }
+
+
+            for chunk_line in range(2,len(chunk_list)):
+                part_line = chunk_list[chunk_line]
+                if len(part_line) == 0:
                     continue
-                part_type = ":".join(split_line[1:]).strip()
-            if part_type is None:
-                continue
-            self.model.lsblk[disk_name]["PARTTABLE"] = part_type
+                part_line_split = part_line.split(':')
+                part_path = disk_line_split[0] + part_line_split[0]
+                part_line_dict = {
+                    'Path' : part_path,
+                    'Number' : part_line_split[0],
+                    'Start' : part_line_split[1],
+                    'End' : part_line_split[2],
+                    'Size' : part_line_split[3],
+                    'File system' : part_line_split[4],
+                    'Flags' : part_line_split[4].split(',')
+                }
+                parted_dict_disk['partition'][part_path] = part_line_dict
+            parted_dict[disk_line_split[0]] = parted_dict_disk
+        self.model.parted = parted_dict
+
+
+    def partitions_all_refresh(self):
```

Comment:
Okay, I see the distinction, but it may be better to drop the partitions_all prefix.  Calling refresh_lsblk() and refresh_parted() seems more obvious to me.